shared/desktop/yarn-helper/font.js

```diff
@@ -23,183 +17,90 @@ const commands = {
-      }
-    }
-  })
+const paths = {
+  iconfont: path.resolve(__dirname, '../../images/iconfont'),
+  iconpng: path.resolve(__dirname, '../../images/icons'),
+  font: path.resolve(__dirname, '../../fonts'),
+  iconConstants: path.resolve(__dirname, '../../common-adapters/icon.constants.js'),
 }
 
-function svgToGridMap() {
-  const grids = {}
-
-  fs.readdirSync(path.resolve(__dirname, '../../images/iconfont')).forEach(i => {
-    const match = i.match(/^kb-iconfont-(.*)-(\d+).svg$/)
-    if (match && match.length === 3) {
-      const name = match[1]
-      const p = path.resolve(path.resolve(__dirname, '../../images/iconfont'), i)
-      const gridSize = match[2]
-
-      if (!grids[gridSize]) {
-        grids[gridSize] = {}
-      }
-
-      grids[gridSize][name] = {gridSize, name, path: p}
-    }
-  })
-
-  return grids
-}
-function generateIcoMoon() {
-  const svgPaths = {}
-  // Need to get the svg info from iconmoon. Couldn't figure out how to derive exactly what they need from the files themselves
-  JSON.parse(
-    fs.readFileSync(path.join(__dirname, '../../images/iconfont/kb-icomoon-project-app.json'), 'utf8')
-  ).icons.forEach(icon => {
-    svgPaths[icon.tags[0]] = icon.paths
-  })
-
-  const grids = svgToGridMap()
-
-  let selectionOrder = 1
-  let selectionID = 1
-
-  const iconSets = Object.keys(grids).map((size, idx) => ({
-    colorThemes: [],
-    height: 1024,
-    icons: Object.keys(grids[size]).map((name, idx) => {
-      const paths = svgPaths[`kb-iconfont-${name}-${size}`]
-      if (!paths) {
-        throw new Error(
-          `Can't find path for ${name}. Did you run the svgs through icomoon and update kb-icomoon-project-app.json?`
-        )
-      }
-      return {
-        attrs: [],
-        grid: size,
-        id: idx,
-        isMulticolor: false,
-        paths,
-        selection: [],
-        tags: [name],
-      }
-    }),
-    id: idx,
-    metadata: {
-      name: `Grid ${size}`,
-    },
-    prevSize: 12,
-    selection: Object.keys(grids[size]).map((name, idx) => ({
-      id: selectionID++,
-      name,
-      order: selectionOrder++,
-      prevSize: size,
-    })),
-  }))
-
-  const write = {
-    iconSets,
-    metadata: {
-      created: 1478124107835,
-      lastOpened: 1478124176910,
-      name: 'KB icon fonts',
-    },
-    preferences: {
-      fontPref: {
-        embed: false,
-        ie7: false,
-        metadata: {
-          fontFamily: 'kb',
-          majorVersion: 1,
-          minorVersion: 0,
+const baseCharCode = 0xe900
+const iconfontRegex = /^(\d+)-kb-iconfont-(.*)-(\d+).svg$/
+const getSvgFilenames = () =>
+  fs
+    .readdirSync(paths.iconfont)
+    .filter(i => i.match(iconfontRegex))
+    // Need to force Node to interpret the counter as a Number so we don't get 1, 10, 100
+    .sort((x, y) => {
+      const matchX = x.match(iconfontRegex)
+      const matchY = y.match(iconfontRegex)
+      if (!matchX || !matchY || matchX.length !== 4 || matchY.length !== 4) return 0
```

Comment:
i think this function could be simpler
```
fs
    .readdirSync(paths.iconfont)
    .map(path => {
        const match = path.match(iconfontRegex)
        return (match && match.length === 4) ? {path, score: match[1]} : null
    })
    .filter(Boolean)
    .sort((x, y) => x.score - y.score)
    .map(({path}) => path.resolve(paths.iconfont, path))

```

Reply:
Just wanted to separate the filenames from the paths because they are needed if different contexts. One is for the icon.constants.js and the other is for passing into webfonts-generator.

Reply:
in the first map you could return {path, match} so you have both

Reply:
and you run the regex once per, not during the sort